File: clients/app/android/app/src/main/java/com/petsitter/ui/components/AppBadge.kt
package com.petsitter.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import com.petsitter.ui.theme.AppColors
import com.petsitter.ui.theme.AppTypography

enum class BadgeVariant { Primary, Success, Warning, Danger, Neutral }

@Composable
fun AppBadge(
    text: String,
    variant: BadgeVariant = BadgeVariant.Neutral,
) {
    val (bg, fg) = when (variant) {
        BadgeVariant.Primary -> AppColors.primaryLight to AppColors.primary
        BadgeVariant.Success -> AppColors.successLight to AppColors.success
        BadgeVariant.Warning -> AppColors.warningLight to AppColors.warning
        BadgeVariant.Danger  -> AppColors.dangerLight  to AppColors.danger
        BadgeVariant.Neutral -> AppColors.grey200      to AppColors.textSecondary
    }

    Box(
        modifier = Modifier
            .background(bg, RoundedCornerShape(100.dp))
            .padding(horizontal = 8.dp, vertical = 3.dp),
    ) {
        Text(text, style = AppTypography.caption, color = fg)
    }
}

File: clients/app/ios/PetSitter/DesignSystem/Atoms/AppBadge.swift
import SwiftUI

enum BadgeVariant { case primary, success, warning, danger, neutral }

struct AppBadge: View {
    let label: String
    var variant: BadgeVariant = .primary

    private var bgColor: Color {
        switch variant {
        case .primary: return AppColors.primaryLight
        case .success: return AppColors.successLight
        case .warning: return AppColors.warningLight
        case .danger:  return AppColors.dangerLight
        case .neutral: return AppColors.grey200
        }
    }
    private var textColor: Color {
        switch variant {
        case .primary: return AppColors.primary
        case .success: return AppColors.success
        case .warning: return AppColors.warning
        case .danger:  return AppColors.danger
        case .neutral: return AppColors.textSecondary
        }
    }

    var body: some View {
        Text(label)
            .font(AppTypography.caption)
            .foregroundStyle(textColor)
            .padding(.horizontal, 8)
            .padding(.vertical, 4)
            .background(bgColor)
            .cornerRadius(4)
    }
}
